lib/onigumo.ex

```diff
@@ -2,19 +2,29 @@ defmodule Onigumo do
   @moduledoc """
   Web scraper
   """
-  @url "https://www.httpbin.org/html"
-  @filename "body.html"
+  @input_filename "urls.txt"
+  @output_filename "body.html"
 
-  def download() do
+  def main() do
     HTTPoison.start()
     http = http_client()
 
+    load_urls(@input_filename)
+    |> Enum.map(&download(http, &1))
+  end
+
+  def download(http_client, url) do
     %HTTPoison.Response{
       status_code: 200,
       body: body
-    } = http.get!(@url)
+    } = http_client.get!(url)
+
+    File.write!(@output_filename, body)
+  end
 
-    File.write!(@filename, body)
+  def load_urls(filepath) do
+    File.stream!(filepath, [:read], :line)
```

Comment:
The `:read` and `:line` part is redundant, but it makes it nicely explicit that we are **reading** the file by **lines**.

Reply:
good point